File: forest/rich-forager.lua
loop
    local flag = self:findFlag()
    local enemy = self:findNearestEnemy()
    local item = self:findNearestItem()
    if flag then
        self:pickUpFlag(flag)
    elseif enemy then
        d = self:distanceTo(enemy)
        if d < 10 and self:isReady("cleave") then
            self:cleave(enemy)
        else
            self:attack(enemy)
        end
    elseif item then
        self:moveXY(item.pos.x, item.pos.y)
    end
end

File: forest/rich-forager.clj
(while true
  (def f (.findFlag this))
  (if f
    (.pickUpFlag this f)
    (when true
      (def e (.findNearestEnemy this))
      (if e
        (when true
          (if (.isReady this "bash")
            (.bash this e)
            (if (.isReady this "cleave")
              (.cleave this e)
              (.attack this e))))
        (when true
          (def i (.findNearestItem this))
          (if i
            (.move this (.pos i))
            (.say this "I'm ready.")))))))
